insolar/record/record.go

```diff
@@ -270,3 +273,23 @@ func CalculateRequestAffinityRef(
 	}
 	return affinityRef
 }
+
+// ObjectIDFromRequest calculates object is from request. Passed hasher must be newly created.
+func ObjectIDFromRequest(hasher hash.Hash, request Request, requestID insolar.ID) (insolar.ID, error) {
```

Comment:
if you need to have fresh hasher, maybe it is better to get ReferenceHasher, that will allow you to create hasher within the function?